hibernate-core/src/main/java/org/hibernate/boot/model/FunctionContributor.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+package org.hibernate.boot.model;
+
+import org.hibernate.query.sqm.function.SqmFunctionRegistry;
+import org.hibernate.service.ServiceRegistry;
+
+/**
+ * Contract for contributing functions
+ *
+ * @author Karel Maesen
+ */
+public interface FunctionContributor {
```

Comment:
I know it is implicitly, but can we mark this as `@FunctionalInterface`?  It helps when refactoring later

Reply:
It's not necessarily. It's a service provider

Reply:
A service provider is implicitly tied to a service contract (here `FunctionContributor`).  So unless you think people will be registering this implementing class under different service contracts, it in can.

But I agree it is probably best to not define it that way